### PR TITLE
Hackathon Participant Dashboard

### LINKED ISSUE #67: [FUP-50] Portal dashboards for hackathon
<h3>Portal dashboard for hackathon</h3><p>A portal view for a hackathon would be needed for:</p><ul class="list-disc list-outside leading-3 -mt-2 tight" data-tight="true"><li class="leading-normal -mb-2"><p>Event admins</p></li><li class="leading-normal -mb-2"><p>Teams</p></li></ul><p></p><p><strong>Teams</strong></p><ul class="not-prose pl-2" data-type="taskList"><li class="flex items-start my-4" data-checked="false" data-type="taskItem"><label><input type="checkbox"><span></span></label><div><p>Create a project - with necessary fields</p></div></li><li class="flex items-start my-4" data-checked="false" data-type="taskItem"><label><input type="checkbox"><span></span></label><div><p>Adding members</p></div></li><li class="flex items-start my-4" data-checked="false" data-type="taskItem"><label><input type="checkbox"><span></span></label><div><p>Project updates and comments</p><p></p></div></li></ul>

### PR DESCRIPTION
**Pull request by harshtandiya**
## Things Done
- Hackathon Team creation workflow
- Project Creation workflow: with new project creation as well as contribution project creation workflow

## Demo:
[Screencast from 2024-06-03 15-00-40.webm](https://github.com/fossunited/fossunited/assets/73123690/54dabe2f-2c11-4ebe-ad4a-c8f85dfeded3)

### CODE REVIEW

File: dashboard/src/pages/hackathon/InitRegister.vue [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          </p>
+        </div>
+      </div>
+      <RouterView/>
+    </div>
+  </div>
+</template>
+<script setup>
+import Header from '@/components/Header.vue'
+import HackathonHeader from '@/components/hackathon/HackathonParitipantHeader.vue'
```
Comment (rahulporuri) [inline dashboard/src/pages/hackathon/InitRegister.vue]:
```suggestion
import HackathonHeader from '@/components/hackathon/HackathonParticipantHeader.vue'
```

**Comment by RiyaMathew-11** [comment]
Meta comments on the workflow on `contribute` : 

1. There could be 5 to 6 partner projects, I think it could be multi-select/radio-button list rather than tiles, takes up less space, and won't be too crowded. 

 - The project details are already mentioned in the look-up page. We could link that page here - add it in the guidelines. 
 
2. Proposing to rename the following fields: 

 - `Repository Link` to `Pull Request Link`
 - `Project Description` - `Describe Contribution`

**Comment by RiyaMathew-11** [comment]
1. Whether they choose to build a project a contribute, both of them would come under submissions in the submission listing page ? 

2. I think we could also add rules in the dashboard homepage ? 

   Rules in a nutshell
    1. Evaluation will be done on basis of code commit during the course of the event
    2. You cannot use external APIs as the core feature
    3. Your project must have a valid [FOSS license](https://opensource.org/licenses)
    4. The cash prize will be split among the winners at the discretion of the jury
    5. Check out the [rules page](https://archive.fossunited.org/fosshack/rules) for more details.

**Comment by harshtandiya** [comment]
@RiyaMathew-11 

- 6, even 9 will be fine imo, let's go with this for now.
- Sorry what look up page? 
- Pull request and issue linking will have their own workflow after team creation

- Yes both would be in submission listing
- Rules are dynamic, can be added through desk
- Can you specify which homepage?

**Comment by RiyaMathew-11** [comment]
> Sorry what look up page?

The listing page for partner projects with links to their issue board and discussion links. (TBD - once projects get finalised)

Projects Page

> 6, even 9 will be fine imo, let's go with this for now.

It could be fine, but it feels a bit redundant, especially when we are trying to give separate space to community and commercial projects. This display with details etc. is quite contrary to that, I think, comparing to a simpler list/dropdown to choose from - But we could adopt anyway. 

> Pull request and issue linking will have their own workflow after team creation

Please explain how you are planning this ? I think it would be better if we keep the workflow simple. 

> Rules are dynamic, can be added through desk
Can you specify which homepage?

Yeah they should be dynamic, we could add them in the team dashbaord page - (second screen in the gif). That would be where a team might spend more time - better visibility of the rules too. 

I have seen these in some hackathon platforms.

**Comment by harshtandiya** [comment]
@RiyaMathew-11 

- Let's link this page in the guidelines section. It can be set up through desk.
- Let's experiment with this, if it is not well received we can revert
- Child tables
- Hackathon Rules right? yeah maybe it would be better to add those in the dashboard

**Pull request merged**